tools/cache_creator/cache_creator.cpp

```diff
@@ -161,6 +162,29 @@ bool hexStringToUuid(llvm::StringRef hexStr, llvm::MutableArrayRef<uint8_t> outU
   return true;
 }
 
+llvm::Expected<ElfLlpcCacheInfo> getCacheInfoFromMetadataBlob(llvm::StringRef noteBlob) {
+  llvm::msgpack::Document *document = new llvm::msgpack::Document;
+  if (!document->readFromBlob(noteBlob, false)) {
+    return llvm::createStringError(llvm::object::object_error::invalid_file_type, "Bad PAL metadata format");
+  }
+
+  llvm::msgpack::MapDocNode pipelineNode =
+      document->getRoot().getMap(true)["amdpal.pipelines"].getArray(true)[0].getMap(true);
+  llvm::msgpack::MapDocNode cacheInfoNode = pipelineNode[".xgl_cache_info"].getMap(true);
+
+  llvm::msgpack::ArrayDocNode hashNode = cacheInfoNode[".128_bit_cache_hash"].getArray(true);
+  Util::MetroHash::Hash hash = {.qwords = {hashNode[0].getUInt(), hashNode[1].getUInt()}};
```

Comment:
Don't use designated initializers, they are a compiler extension in C++14. You can zero initialize the hash first and then set two fields separately.
```
Hash hash = {};
hash.qwords[0] = ...;
hash.qwords[1] = ...;
```